packages/@ember/-internals/glimmer/lib/components/link-to.ts

```diff
@@ -732,16 +733,8 @@ const LinkComponent = EmberComponent.extend({
     let resolvedQueryParams = {};
     let queryParams = get(this, 'queryParams');
 
-    if (!queryParams) {
-      return resolvedQueryParams;
-    }
-
-    let values = queryParams.values;
-    for (let key in values) {
-      if (!values.hasOwnProperty(key)) {
-        continue;
-      }
-      resolvedQueryParams[key] = values[key];
+    if (queryParams) {
+      resolvedQueryParams = assign(resolvedQueryParams, queryParams.values);
```

Comment:
I don’t think we need assignment here since we are mutating `resolvedQueryParams`